graphql/database/users/getShouldShowYahooPrompt.js

```diff
@@ -15,9 +17,14 @@ const getShouldShowYahooPrompt = async (userContext, user) => {
     user,
     YAHOO_SEARCH_NEW_USERS
   )
+  const alreadyResponded =
+    user.yahooSearchSwitchPrompt &&
+    user.yahooSearchSwitchPrompt.hasRespondedToPrompt
+  const alreadyUsingSearchEngine =
+    user.searchEngine === charitableSearchEngineId
   return (
-    !user.yahooSwitchSearchPrompt &&
-    !user.yahooPaidSearchRewardOptIn &&
+    !alreadyResponded &&
+    !alreadyUsingSearchEngine &&
```

Comment:
How come there's no more `yahooPaidSearchRewardOptIn` being checked here?

Reply:
@jedtan I figure what matters here is if the user's not using the charitable search engine. If they're **not** using the charitable search engine but previously opted in, we still want to ask them to switch search. Does my thinking make sense?

Reply:
My bad: doing this broke the intended behavior for the new users experiment. Will revert.